src/test/java/org/terasology/itempipes/ItemPipesTest.java

```diff
@@ -127,23 +145,27 @@ public void chestInputTest() {
 
     @Test
     public void minimumVelocityTest() {
-        Vector3i newPipeLoc = Vector3i.zero();
-        placeBlock(new Vector3i(newPipeLoc), "ItemPipes:basicPipe");
-        newPipeLoc.add(Vector3i.up());
-        placeBlock(new Vector3i(newPipeLoc), "ItemPipes:basicPipe");
-        newPipeLoc.add(Vector3i.east());
-        placeBlock(new Vector3i(newPipeLoc), "ItemPipes:basicPipe");
-        newPipeLoc.add(Vector3i.down());
-        placeBlock(new Vector3i(newPipeLoc), "ItemPipes:basicPipe");
-
-        EntityRef droppedItem = dropBlockItem(Vector3f.west().add(Vector3f.up()), "ItemPipes:suction");
-        EntityRef startPipe = blockEntityRegistry.getBlockEntityAt(Vector3i.zero());
+        this.initialize();
+
+        Vector3i newPipeLoc = new Vector3i();
+        placePipe(new Vector3i(newPipeLoc));
```

Comment:
this one has a name!

and it's also duplicated to a new instance every time? hmm.

I know there's a category of bug that could be lurking here: if code is comparing these by their identity instead of their values.

but if that's the sort of bug it's trying to cover in the tests, I'd suggest making a test explicitly for that, instead of some undocumented characteristic of a `minimumVelocityTest`.

(this is out-of-scope for this PR, as I know you're just porting the tests as-is. But maybe it informs how you do pass `newPipeLoc` when you add it to the other tests above.)

Reply:
the vector is modified between the calls to `placePipe`, and although `placePipe` guarantees it is only reading the vector by accepting a constant `Vector3ic` it _might_ store the value for later usage, and could potentially see a different value then...

I think in this case it should be fine to not create new instances all the time...

Reply:
oh, yes, that `newPipeLoc.add` does modify that instance, I see :+1: